### PR TITLE
Added pagination to home.html

### PR DESCRIPTION
**Pull request by aterenin**
This PR adds pagination support to Minima.

To enable, set `paginate: 5` in `_config.yml` and run Jekyll.

### CODE REVIEW

**Comment by ashmaroli** [comment]
Thank you for your contribution, @aterenin 
Any associated styling for the pagination?

IMO, it'd be better if the code for pagination be abstracted to an `include`. That way it could easily used in another layout as well..

How about some addition to the documentation? It'd be better to note that only `jekyll-paginate` is supported by GitHub Pages. The successor `jekyll-paginate-v2` isn't..

File: _layouts/home.html [hunk 1]
```diff
@@ -6,12 +6,18 @@
...
+  {% else %}
+  {% assign posts = site.posts %}
+  {% endif %}
+
   <h1 class="page-heading">Posts</h1>
-  
+
   <ul class="post-list">
-    {% for post in site.posts %}
+    {% assign date_format = site.minima.date_format | default: "%b %-d, %Y" %}
```
Comment (ashmaroli) [inline _layouts/home.html]: let's handle moving this `assign` in another PR..

File: _layouts/home.html [hunk 2]
```diff
@@ -6,12 +6,18 @@
 
   {{ content }}
 
+  {% if site.paginate %}
+  {% assign posts = paginator.posts %}
+  {% else %}
+  {% assign posts = site.posts %}
+  {% endif %}
```
Comment (ashmaroli) [inline _layouts/home.html]: Would the following work?
```
{% assign posts = site.posts %}
{% if site.paginate %}
  {% assign posts = paginator.posts %}
{% endif %}
```

**Comment by aterenin** [comment]
Requested changes have been made.

I am not sure what the right way to abstract it would be. There's essentially two parts: one which generates a list of posts, and one which generates `Page 1 of 5`. I do not see how the same code would be used elsewhere. Is there a good example I may not have considered?

I'm not sure precisely how Github Pages handles pagination. In my own site, which uses this code, I do not need to enable any plugins for pagination - just adding the line `paginate: 5` suffices. I'll update the readme file to indicate this.

**Comment by pathawks** [comment]
I think this is handled with the **jekyll-paginate** plugin, which [GitHub Pages includes](https://pages.github.com/versions/).

**Comment by aterenin** [comment]
Wasn't aware GH pages includes that - it isn't in my `_config.yml` but I presume since the code works that it still gets loaded.

Regarding the two separate plugins - `jekyll-paginate` and `jekyll-paginate-v2` - my code should seamlessly support both, as it checks for existence of a `paginator` object rather than a specific plugin. I added a link to the readme to point toward the relevant Jekyll page, since many people will be using GH pages.

**Pull request closed without merging**